Automattic/wp-calypso
# A4A: Add base code for the Hosting tab at the Preview Pane

Resolves: https://github.com/Automattic/automattic-for-agencies-dev/issues/262

## Proposed Changes

This PR adds the new Hosting tab to the Preview Pane as a code base with a draft Hosting tab content.

The Hosting tab is behind this feature flag: `a4a/hosting-preview-pane` and active in `development` and `horizon`.

![image](https://github.com/Automattic/wp-calypso/assets/9832440/0244c729-0005-40c4-8b71-19dfeb1981dc)


Also, this PR reorganizes the folder/files a bit:

- `features/hosting/overview.tsx` - This is the Hosting Overview Preview Pane with the Hosting provider, PHP/WP version. The `hosting` folder represent the Hosting family context, and here we will add the different tabs the Hosting family will have. This PR adds the first tab, the Hosting Overview content.
- `features/jetpack/jetpack-preview-pane` - This will be the Jetpack family Preview Pane. I cleaned up it and left only Jetpack features, and I removed the A4A Details tabs, moving it to the A4A Overview family.
- `features/overview/index.tsx` moved to `features/a4a/overview-preview-pane.tsx` - This is the entry point. Now, is where we build and add the feature tabs (Jetpack, Hosting, and Details) to the Preview Pane.

## Testing Instructions

- Check the code and become familiar with the feature folder structure.
- You have to see the new Hosting tab, and its content (draft content)
- Navigate through the tabs to ensure that everything is working
- The Hosting tab should have this slug/route: /sites/overview/{site_url}/hosting-overview

## Pre-merge Checklist
